doxygen/man/man3/tripPlanner.3
.TH "tripPlanner" 3 "Mon May 11 2020" "Version 1" "CS1D - Project2 - Baseball" \" -*- nroff -*-
.ad l
.nh
.SH NAME
tripPlanner
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <tripplanner\&.h>\fP
.PP
Inherits QDialog\&.
.SS "Public Slots"

.in +1c
.ti -1c
.RI "void \fBonTypologyClick\fP ()"
.br
.RI "Displays stadiums by ballpark topology in MLB information tab\&. "
.ti -1c
.RI "void \fBonAmericanClick\fP ()"
.br
.RI "Displays stadiums in the American League in MLB information tab\&. "
.ti -1c
.RI "void \fBonShowClick\fP ()"
.br
.RI "Displays all stadiums in MLB information tab\&. "
.ti -1c
.RI "void \fBonNationalClick\fP ()"
.br
.RI "Displays stadiums in the National League in MLB information tab\&. "
.ti -1c
.RI "void \fBonStadiumsClick\fP ()"
.br
.RI "Displays teams and stadiums sorted by stadium name in MLB information tab\&. "
.ti -1c
.RI "void \fBonTeamClick\fP ()"
.br
.RI "Displays teams and stadiums sorted by team name in MLB information tab\&. "
.ti -1c
.RI "void \fBonSearchClick\fP ()"
.br
.RI "Displays team that matches searched team name in MLB information tab\&. "
.ti -1c
.RI "void \fBonRoofClick\fP ()"
.br
.RI "Displays stadiums by roof type in MLB information tab\&. "
.ti -1c
.RI "void \fBonDateClick\fP ()"
.br
.RI "Displays stadiums by creation year in MLB information tab\&. "
.ti -1c
.RI "void \fBonSeatingClick\fP ()"
.br
.RI "Displays stadiums by number of seats in MLB information tab\&. "
.ti -1c
.RI "void \fBonGreatClick\fP ()"
.br
.RI "Displays stadiums by largest distance field in MLB information tab\&. "
.ti -1c
.RI "void \fBonLeastClick\fP ()"
.br
.RI "Displays stadiums by shortest distance field in MLB information tab\&. "
.ti -1c
.RI "void \fBinitializeList\fP ()"
.br
.RI "Initalizes checkbox list\&. "
.ti -1c
.RI "void \fBonPlanClick\fP ()"
.br
.RI "Plans trip path requires a valid tripID entered uses the algorithm function to determine path through stadiums requires a valid starting stadium\&. "
.ti -1c
.RI "void \fBonGraphInfoClick\fP ()"
.br
.RI "Opens a dialogue displaying the BFS, DFS, and MST distances of the stadium graph\&. "
.ti -1c
.RI "void \fBonStartTrip\fP ()"
.br
.RI "Initiates the start of the trip requires valid trip plan and ID\&. "
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBtripPlanner\fP (QWidget *parent=nullptr)"
.br
.RI "default explicit Constructor "
.ti -1c
.RI "\fB~tripPlanner\fP ()"
.br
.RI "Destructor\&. "
.ti -1c
.RI "void \fBupdateCombo\fP ()"
.br
.RI "Updates combo box in MLB information\&. "
.ti -1c
.RI "void \fBupdateDistTable\fP ()"
.br
.RI "Updates the table in the Distances tab\&. "
.ti -1c
.RI "void \fBupdateMLBTable\fP ()"
.br
.RI "Updates the table in the MLB information tab\&. "
.ti -1c
.RI "void \fBselectedCollegeList\fP ()"
.br
.RI "Creates the checkbox list of stadiums\&. "
.ti -1c
.RI "bool \fBcollegeDoesExist\fP (QString colName)"
.br
.RI "Checks if a stadium name exists\&. "
.ti -1c
.RI "bool \fBplanDoesExist\fP (QString colName)"
.br
.RI "Checks if the given trip ID already exists\&. "
.ti -1c
.RI "void \fBshowTrip\fP (QString ID)"
.br
.RI "Displays the trip order of the stadiums\&. "
.ti -1c
.RI "void \fBupdateComboTwo\fP ()"
.br
.RI "Updates teh combo box for available stadiums\&. "
.ti -1c
.RI "void \fBalgorithm\fP ()"
.br
.RI "Trip planning algorithm using dijkstra\&. "
.in -1c
.SS "Private Slots"

.in +1c
.ti -1c
.RI "void \fBChecboxChanged\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "Ui::tripPlanner * \fBui\fP"
.br
.RI "points to \fBtripPlanner\fP UI "
.ti -1c
.RI "\fBDbManager\fP \fBmyDb\fP = \fBDbManager\fP(PROJECT_PATH + '/college\&.db')"
.br
.RI "assigns DB to predefined PROJECT_PATH constant "
.ti -1c
.RI "\fBtripprogress\fP * \fBprogress\fP"
.br
.ti -1c
.RI "QVector< QCheckBox * > \fBcheckBoxVector\fP"
.br
.ti -1c
.RI "QStringList \fBselectedColleges\fP"
.br
.ti -1c
.RI "QStringList \fBplannedColleges\fP"
.br
.ti -1c
.RI "int \fBdistance\fP"
.br
.ti -1c
.RI "QString \fBid\fP"
.br
.ti -1c
.RI "QList< int > \fBdistanceTo\fP"
.br
.ti -1c
.RI "\fBgraphInfo\fP * \fBinfo\fP"
.br
.ti -1c
.RI "vector< QString > \fBpath\fP"
.br
.ti -1c
.RI "vector< QString > \fBremainingChoices\fP"
.br
.ti -1c
.RI "vector< int > \fBdistNext\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 16 of file tripplanner\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "tripPlanner::tripPlanner (QWidget * parent = \fCnullptr\fP)\fC [explicit]\fP"

.PP
default explicit Constructor Constructor sets up ui for \fBtripPlanner\fP and executes initalizeList(), and \fBupdateCombo()\fP 
.PP
\fBParameters\fP
.RS 4
\fI*parent\fP - for QDialog 
.RE
.PP

.PP
Definition at line 7 of file tripplanner\&.cpp\&.
.PP
References initializeList(), updateCombo(), updateComboTwo(), updateDistTable(), and updateMLBTable()\&.
.SS "tripPlanner::~tripPlanner ()"

.PP
Destructor\&. Close the \fBtripPlanner\fP UI 
.PP
Definition at line 21 of file tripplanner\&.cpp\&.
.PP
References ui\&.
.SH "Member Function Documentation"
.PP 
.SS "void tripPlanner::algorithm ()"

.PP
Trip planning algorithm using dijkstra\&. 
.PP
Definition at line 627 of file tripplanner\&.cpp\&.
.PP
References algorithm()\&.
.PP
Referenced by algorithm(), and onPlanClick()\&.
.SS "void tripPlanner::ChecboxChanged ()\fC [private]\fP, \fC [slot]\fP"

.PP
Definition at line 406 of file tripplanner\&.cpp\&.
.SS "bool tripPlanner::collegeDoesExist (QString colName)"

.PP
Checks if a stadium name exists\&. 
.PP
Definition at line 547 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::initializeList ()\fC [slot]\fP"

.PP
Initalizes checkbox list\&. 
.PP
Definition at line 367 of file tripplanner\&.cpp\&.
.PP
Referenced by tripPlanner()\&.
.SS "void tripPlanner::onAmericanClick ()\fC [slot]\fP"

.PP
Displays stadiums in the American League in MLB information tab\&. 
.PP
Definition at line 51 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onDateClick ()\fC [slot]\fP"

.PP
Displays stadiums by creation year in MLB information tab\&. 
.PP
Definition at line 277 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onGraphInfoClick ()\fC [slot]\fP"

.PP
Opens a dialogue displaying the BFS, DFS, and MST distances of the stadium graph\&. 
.PP
Definition at line 618 of file tripplanner\&.cpp\&.
.PP
References info, and graphInfo::onCloseClick()\&.
.SS "void tripPlanner::onGreatClick ()\fC [slot]\fP"

.PP
Displays stadiums by largest distance field in MLB information tab\&. 
.PP
Definition at line 321 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onLeastClick ()\fC [slot]\fP"

.PP
Displays stadiums by shortest distance field in MLB information tab\&. 
.PP
Definition at line 344 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onNationalClick ()\fC [slot]\fP"

.PP
Displays stadiums in the National League in MLB information tab\&. 
.PP
Definition at line 98 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onPlanClick ()\fC [slot]\fP"

.PP
Plans trip path requires a valid tripID entered uses the algorithm function to determine path through stadiums requires a valid starting stadium\&. 
.PP
Definition at line 472 of file tripplanner\&.cpp\&.
.PP
References algorithm(), distance, and selectedCollegeList()\&.
.SS "void tripPlanner::onRoofClick ()\fC [slot]\fP"

.PP
Displays stadiums by roof type in MLB information tab\&. 
.PP
Definition at line 254 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onSearchClick ()\fC [slot]\fP"

.PP
Displays team that matches searched team name in MLB information tab\&. 
.PP
Definition at line 227 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onSeatingClick ()\fC [slot]\fP"

.PP
Displays stadiums by number of seats in MLB information tab\&. 
.PP
Definition at line 298 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onShowClick ()\fC [slot]\fP"

.PP
Displays all stadiums in MLB information tab\&. 
.PP
Definition at line 76 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onStadiumsClick ()\fC [slot]\fP"

.PP
Displays teams and stadiums sorted by stadium name in MLB information tab\&. 
.PP
Definition at line 124 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onStartTrip ()\fC [slot]\fP"

.PP
Initiates the start of the trip requires valid trip plan and ID\&. 
.PP
Definition at line 666 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onTeamClick ()\fC [slot]\fP"

.PP
Displays teams and stadiums sorted by team name in MLB information tab\&. 
.PP
Definition at line 145 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::onTypologyClick ()\fC [slot]\fP"

.PP
Displays stadiums by ballpark topology in MLB information tab\&. 
.PP
Definition at line 26 of file tripplanner\&.cpp\&.
.SS "bool tripPlanner::planDoesExist (QString colName)"

.PP
Checks if the given trip ID already exists\&. 
.PP
Definition at line 561 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::selectedCollegeList ()"

.PP
Creates the checkbox list of stadiums\&. 
.PP
Definition at line 442 of file tripplanner\&.cpp\&.
.PP
Referenced by onPlanClick()\&.
.SS "void tripPlanner::showTrip (QString ID)"

.PP
Displays the trip order of the stadiums\&. 
.PP
Definition at line 574 of file tripplanner\&.cpp\&.
.SS "void tripPlanner::updateCombo ()"

.PP
Updates combo box in MLB information\&. 
.PP
Definition at line 166 of file tripplanner\&.cpp\&.
.PP
Referenced by tripPlanner()\&.
.SS "void tripPlanner::updateComboTwo ()"

.PP
Updates teh combo box for available stadiums\&. 
.PP
Definition at line 598 of file tripplanner\&.cpp\&.
.PP
Referenced by tripPlanner()\&.
.SS "void tripPlanner::updateDistTable ()"

.PP
Updates the table in the Distances tab\&. 
.PP
Definition at line 187 of file tripplanner\&.cpp\&.
.PP
Referenced by tripPlanner()\&.
.SS "void tripPlanner::updateMLBTable ()"

.PP
Updates the table in the MLB information tab\&. 
.PP
Definition at line 206 of file tripplanner\&.cpp\&.
.PP
Referenced by tripPlanner()\&.
.SH "Member Data Documentation"
.PP 
.SS "QVector<QCheckBox*> tripPlanner::checkBoxVector\fC [private]\fP"

.PP
Definition at line 203 of file tripplanner\&.h\&.
.SS "int tripPlanner::distance\fC [private]\fP"

.PP
Definition at line 208 of file tripplanner\&.h\&.
.PP
Referenced by onPlanClick()\&.
.SS "QList<int> tripPlanner::distanceTo\fC [private]\fP"

.PP
Definition at line 212 of file tripplanner\&.h\&.
.SS "vector<int> tripPlanner::distNext\fC [private]\fP"

.PP
Definition at line 220 of file tripplanner\&.h\&.
.SS "QString tripPlanner::id\fC [private]\fP"

.PP
Definition at line 210 of file tripplanner\&.h\&.
.SS "\fBgraphInfo\fP* tripPlanner::info\fC [private]\fP"

.PP
Definition at line 214 of file tripplanner\&.h\&.
.PP
Referenced by onGraphInfoClick()\&.
.SS "\fBDbManager\fP tripPlanner::myDb = \fBDbManager\fP(PROJECT_PATH + '/college\&.db')\fC [private]\fP"

.PP
assigns DB to predefined PROJECT_PATH constant 
.PP
Definition at line 199 of file tripplanner\&.h\&.
.SS "vector<QString> tripPlanner::path\fC [private]\fP"

.PP
Definition at line 216 of file tripplanner\&.h\&.
.SS "QStringList tripPlanner::plannedColleges\fC [private]\fP"

.PP
Definition at line 207 of file tripplanner\&.h\&.
.SS "\fBtripprogress\fP* tripPlanner::progress\fC [private]\fP"

.PP
Definition at line 201 of file tripplanner\&.h\&.
.SS "vector<QString> tripPlanner::remainingChoices\fC [private]\fP"

.PP
Definition at line 218 of file tripplanner\&.h\&.
.SS "QStringList tripPlanner::selectedColleges\fC [private]\fP"

.PP
Definition at line 205 of file tripplanner\&.h\&.
.SS "Ui::tripPlanner* tripPlanner::ui\fC [private]\fP"

.PP
points to \fBtripPlanner\fP UI 
.PP
Definition at line 193 of file tripplanner\&.h\&.
.PP
Referenced by ~tripPlanner()\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for CS1D - Project2 - Baseball from the source code\&.


tripplanner.cpp
#include "tripplanner.h"
#include "ui_tripplanner.h"
#include <vector>
#include <QGraphicsView>
#include <QGraphicsScene>
#include <QMovie>

#include "QVBoxLayout"

tripPlanner::tripPlanner(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::tripPlanner)
{
    ui->setupUi(this);
    updateCombo();
    updateComboTwo();
    updateDistTable();
    updateMLBTable();
    initializeList();
    ui->startTrip->setEnabled(false);

    QMovie *movie = new QMovie(":/img/giphy (1).gif");
    ui->movieLabel->setMovie(movie);
    movie->start();

    QMovie *movie1 = new QMovie(":/img/giphy (2).gif");
    ui->label_3->setMovie(movie1);
    movie1->start();
}

tripPlanner::~tripPlanner()
{
    delete ui;
}

void tripPlanner::onTypologyClick()
{
    QString collegeName = "National";
    qDebug() << collegeName;

    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName, Ballpark FROM MLB ORDER BY Ballpark ASC");


    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();

}

void tripPlanner::onAmericanClick()
{
    QString collegeName = "American";
    qDebug() << collegeName;

    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT * FROM MLB WHERE League = (:League) ORDER BY TeamName ASC");

    qry->bindValue(":League", collegeName);

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();
}

void tripPlanner::onShowClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT * FROM MLB");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();


}

void tripPlanner::onNationalClick()
{
    QString collegeName = "National";
    qDebug() << collegeName;

    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT * FROM MLB WHERE League = (:League) ORDER BY StadiumName ASC");

    qry->bindValue(":League", collegeName);

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();

}

void tripPlanner::onStadiumsClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName FROM MLB ORDER BY StadiumName ASC");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();

}

void tripPlanner::onTeamClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName FROM MLB ORDER BY TeamName ASC");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();

}

void tripPlanner::updateCombo()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT DISTINCT TeamName FROM MLB");

    if(qry->exec())
    {
        qDebug() << "college1 table updated.";
    }
    else
        qDebug() << "failed";

    model->setQuery(*qry);

    ui->comboBox->setModel(model);
}


void tripPlanner::updateDistTable()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT * FROM Distances");

    if(qry->exec())
    {
        qDebug() << "college table updated.";
    }

    model->setQuery(*qry);

    ui->collegeView->setModel(model);
    ui->collegeView->resizeColumnsToContents();
}

void tripPlanner::updateMLBTable()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT * FROM MLB");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();

}

void tripPlanner::onSearchClick()
{
    QString collegeName = this->ui->comboBox->currentText();
    qDebug() << collegeName;

    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT * FROM MLB WHERE TeamName = (:TeamName)");

    qry->bindValue(":TeamName", collegeName);

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();


}

void tripPlanner::onRoofClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName, RoofType FROM MLB WHERE RoofType = (:RoofType) ORDER BY TeamName ASC");
    qry->bindValue(":RoofType", "Open");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();


}

void tripPlanner::onDateClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName, DataOpened FROM MLB ORDER BY DataOpened ASC");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();

}

void tripPlanner::onSeatingClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();
    QSqlQuery* tqry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName, SeatingCapacity FROM MLB ORDER BY SeatingCapacity ASC");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();


}

void tripPlanner::onGreatClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName, max(DistanceField) FROM MLB ");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();



}

void tripPlanner::onLeastClick()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT TeamName, StadiumName, min(DistanceField) FROM MLB ");

    if(qry->exec())
    {
        qDebug() << "souvenir table updated.";
    }

    model->setQuery(*qry);

    ui->souvenirView->setModel(model);
    ui->souvenirView->setModel(model);
    ui->souvenirView->resizeColumnsToContents();



}

void tripPlanner::initializeList()
{
    QSqlQuery *query = new QSqlQuery();

    checkBoxVector.clear();

    QWidget *container = new QWidget;
    QVBoxLayout *vBoxLayout = new QVBoxLayout;

    container->setLayout(vBoxLayout);

    ui->scrollArea->setWidget(container);



    query->prepare("SELECT DISTINCT begStadium FROM Distances");
    if(!query->exec())
    {
        qDebug() << "tripPlanner initializeList query failed";


    }
    else
    {
        while(query->next())
        {
            QCheckBox* checkBox = new QCheckBox(query->value(0).toString());
            checkBox->setCheckState(Qt::CheckState::Unchecked);
            checkBoxVector.push_back(checkBox);
            connect(checkBox, &QCheckBox::stateChanged, this, &tripPlanner::ChecboxChanged);
        }

        for(int i = 0; i < checkBoxVector.size(); i++)
        {
            vBoxLayout->addWidget(checkBoxVector[i]);
        }
    }
}

void tripPlanner::ChecboxChanged()
{

    qDebug() << "Signal caught";

    int checkedCount = 0;

    for(int i = 0; i < checkBoxVector.size(); i++)
    {
        if(checkBoxVector[i]->checkState() == Qt::CheckState::Checked)
        {
            checkedCount++;
        }
    }

    if(checkedCount == 11000)
    {
        for(int i = 0; i < checkBoxVector.size(); i++)
        {
            if(checkBoxVector[i]->checkState() == Qt::CheckState::Unchecked)
            {
                checkBoxVector[i]->setDisabled(true);
            }
            qDebug() << checkBoxVector[i] << endl;
        }
    }
    else
    {
        for(int i = 0; i < checkBoxVector.size(); i++)
        {
            checkBoxVector[i]->setDisabled(false);
        }
    }
}


void tripPlanner::selectedCollegeList()// creates a list of the selected colleges from the checked boxes
{
    selectedColleges.clear();
    QSqlQuery *query = new QSqlQuery();
    int i = 0;

    query->prepare("SELECT DISTINCT begStadium FROM Distances");
    if(!query->exec())
    {
        qDebug() << "tripPlanner initializeList query failed";


    }
    else
    {
        while(query->next())
        {
            if(checkBoxVector[i]->checkState() == Qt::CheckState::Checked)
            {
                QString temp = query->value("begStadium").toString();
                selectedColleges << temp;
                qDebug() << temp;
            }
            i++;
        }

    }

}

void tripPlanner::onPlanClick()
{
    distance = 0;
    QString startingStadium;
    QString tripID;
    startingStadium = this->ui->colName->currentText();
    tripID = this->ui->trip->text();

    selectedCollegeList();

    for(int i = 0; i < selectedColleges.size(); i++)
    {
        if(selectedColleges[i] != startingStadium)
        {
            remainingChoices.push_back(selectedColleges[i]);
        }
    }
    path.push_back(startingStadium);


    if(!collegeDoesExist(startingStadium)) // checks if the user input a correct stadium
    {
        this->ui->warningLabel->setText("Please enter a college that you have selected on the left!");
    }
    else
    {

        qDebug() << myDb.tripIdExists(tripID);
        qDebug() << tripID;
        if(tripID.size() == 3 && !myDb.tripIdExists(tripID))
        {
            this->ui->warningLabel->setText("");

            plannedColleges.clear();
            id = tripID;

            algorithm();

            distanceTo << 0; // adds 0 for distance to next college at last college

            myDb.reOpen();

            for(int index = 0; index < path.size(); index++)
            {
                myDb.addTrip(tripID, path[index], index, distNext[index]); // uploads trip to DB
            }


            qDebug() << path;

            showTrip(tripID);

            QString out = QString::number(distance);
            ui->startTrip->setEnabled(true);


        }
        else
        {
            qDebug() << "I AM HERE//idk ";
        }

    }

    for(int i = 0; i < plannedColleges.size(); i++)
    {
        qDebug() << plannedColleges[i];
    }

    path.clear();
    remainingChoices.clear();
    distNext.clear();

}

bool tripPlanner::collegeDoesExist(QString colName)// checks if a college is added to the trip
{
    for(int i = 0; i < selectedColleges.size(); i++)
    {
        if(colName == selectedColleges[i]) // only going once!!!!
        {
            qDebug() << "true";
            return true;
        }
    }

    return false;
}

bool tripPlanner::planDoesExist(QString colName) // checks if a college is in the planned list
{
    for(int i = 0; i < plannedColleges.size(); i++)
    {
        if(colName == plannedColleges[i])
        {
            return true;
        }
    }

    return false;
}

void tripPlanner::showTrip(QString ID)
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();
    qDebug() << ID;

    qry->prepare("SELECT stadium FROM Trips WHERE tripID = (:tripId)");
    qry->bindValue(":tripId" , ID);
    if(qry->exec())
    {
        qDebug() << "trip table updated.";
    }
    else
        qDebug() << "failed trip table";

    model->setQuery(*qry);

    //clear here?
    ui->window->setModel(new QSqlQueryModel());
    ui->window->setModel(model);
    ui->window->setColumnWidth(20, 400);
}

void tripPlanner::updateComboTwo()
{
    QSqlQueryModel* model=new QSqlQueryModel();

    QSqlQuery* qry=new QSqlQuery();

    qry->prepare("SELECT DISTINCT begStadium FROM Distances");

    if(qry->exec())
    {
        qDebug() << "college1 table updated.";
    }
    else
        qDebug() << "failed";

    model->setQuery(*qry);

    ui->team1->setModel(model);
    ui->team2->setModel(model);
    ui->colName->setModel(model);
}

void tripPlanner::onGraphInfoClick()
{
    info = new graphInfo;

    info->show();

}


void tripPlanner::algorithm()// start is the user selected starting college
{
    QStringList local;
    int closestDist = 100000;
    int closestIndex;
    int tempDist;

    if(remainingChoices.size() > 0)
    {
        for(int i = 0; i < remainingChoices.size(); i++)
        {
            local << path.back() << remainingChoices[i];
            tempDist = performDijkstra(local);
            if(tempDist < closestDist)
            {
                closestDist = tempDist;
                closestIndex = i;
            }

            local.clear();
        }

        distNext.push_back(closestDist);
        qDebug() << closestDist;
        path.push_back(remainingChoices.at(closestIndex));
        qDebug() << remainingChoices.at(closestIndex);
        remainingChoices.erase(remainingChoices.begin() + closestIndex);

        algorithm();

    }
    else
    {
        distNext.push_back(0);
    }


}

void tripPlanner::onStartTrip()
{
    this->hide();
    progress = new tripprogress(this, id);

    progress->show();

}

void tripPlanner::on_pushButton_9_clicked()
{
    QStringList pass;

    pass << ui->team1->currentText() << ui->team2->currentText();

   int temp = performDijkstra(pass);

   QString text = "Distance: " + QString::number(temp);
   ui->distanceLabel->setText(text);

}
